pyqtgraph/opengl/GLViewWidget.py

```diff
@@ -99,10 +98,12 @@ def reset(self):
 
     def addItem(self, item):
         self.items.append(item)
-        if hasattr(item, 'initializeGL'):
+
+        if self.isValid():
             self.makeCurrent()
             try:
                 item.initializeGL()
+                item.setInitialized()
```

Comment:
Are both required? I see none of the subclasses call the parent `initializeGL` since it doesn't do anything. Maybe `setInitialized` could call `initializeGL`? So `GLGraphicsItem.initializeGL` is the user-facing method (consistent with Qt) and the base class handles the state variable for internal usage. In that case `initialize` and `isInitialized` might be clearer.

Reply:
That's a great suggestion, looks less ugly now.